lib/src/patch.dart

```diff
@@ -22,8 +23,9 @@ class Patch {
 
   /// Applies a series of non-overlapping patches to the text of a file.
   static String applyAll(SourceFile file, List<Patch> patches) {
-    var sortedPatches = patches.toList()
-      ..sort((a, b) => a.selection.compareTo(b.selection));
+    var sortedPatches = patches.toList();
+    mergeSort(sortedPatches,
```

Comment:
Add a comment here explaining why you're using `mergeSort()` over `List.sort()`.